In [7]:
import pandas as pd

In [10]:
#Read data from csv file
df = pd.read_csv("F:\\python\Data_Visualization_with_Python_s2-main\\ODI_cricket.csv")
#To display the first 10 rows of the dataframe
display(df.head(10))

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1             Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2            Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3          WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
5       SM Pollock (Afr/ICC/SA)  1996-2008  303   297  15712   9631   393   
6          GD McGrath (AUS/ICC)  1993-2007  250   248  12970   8391   381   
7                   B Lee (AUS)  2000-2012  221   217  11185   8877   380   
8               SL Malinga (SL)  2004-2019  226   220  10936   9760   338   
9         A Kumble (Asia/INDIA)  1990-2007  271   265  14496  10412   337   

     Ave  Econ    SR   4   5  
0  23.08  3.93  35.2  15  10  
1  23.52  3.89  36.2  17   6  
2  23.84  4.68  30.5  14  13  
3  27.53  4.18  39.4   9   4  
4  34.51  4.62  44.7   4   9  
5  24.50  3.67  39.9  12   5  
6  22.02  3.88  34.0   9   7  
7  23.36  4.76  29.4  14   9  
8  28.87  5.35  32.3  11   8  
9  30.89  4.30  43.0   8   2

These first 10 represented output data row show that M Muralitharan played highest matches but Wasim Akram played most innings.In terms of runs we can see 13632 runs that is the highest that was done by Shahid Afridi.18811 is the highests balls done by M Muralitharan but max wickets was taken by him also. Max average runs was done by Shahid Afridi and max strike rate was 44.7 done by him also.Max 4 was done by Wasim Akeam and max 5 was done by Waqar Younis. 

In [11]:
# To find the number of rows and columns in the dataframe
print("number of rows = ", df.shape[0])
print("number of columns = ", df.shape[1])

number of rows =  77
number of columns =  12


In [15]:
 #To find the data statistics and check for the data types
display(df.describe())
print(type(df))

Mat        Inns         Balls          Runs        Wkts  \
count   77.000000   77.000000     77.000000     77.000000   77.000000   
mean   194.402597  181.194805   8839.402597   6671.714286  233.805195   
std     82.485606   67.958393   3316.055457   2245.839029   84.406603   
min     80.000000   76.000000   4074.000000   2821.000000  151.000000   
25%    136.000000  128.000000   6182.000000   5058.000000  173.000000   
50%    170.000000  164.000000   8054.000000   6192.000000  199.000000   
75%    227.000000  218.000000  10750.000000   8021.000000  272.000000   
max    463.000000  372.000000  18811.000000  13632.000000  534.000000   

             Ave       Econ         SR          4          5  
count  77.000000  77.000000  77.000000  77.000000  77.000000  
mean   28.958052   4.596753  37.909091   6.350649   2.870130  
std     4.826768   0.515814   6.060901   3.556929   2.530606  
min    18.680000   3.300000  26.100000   1.000000   0.000000  
25%    24.970000   4.280000  33.000000   3.000000   1.000000  
50%    29.290000   4.660000  37.800000   6.000000   2.000000  
75%    31.900000   4.920000  41.400000   8.000000   4.000000  
max    44.480000   5.830000  52.500000  17.000000  13.000000

<class 'pandas.core.frame.DataFrame'>


In [41]:
# Finding missing values present in the dataset
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Player        77 non-null     object 
 1   Span          77 non-null     object 
 2   Match         77 non-null     int64  
 3   Innings       77 non-null     int64  
 4   Balls         77 non-null     int64  
 5   Runs          77 non-null     int64  
 6   Wickets       77 non-null     int64  
 7   Average       77 non-null     float64
 8   economy_rate  77 non-null     float64
 9   Strike_Rate   77 non-null     float64
 10  Four          77 non-null     int64  
 11  Five runs     77 non-null     int64  
 12  Player_Name   77 non-null     object 
 13  Country       77 non-null     object 
dtypes: float64(3), int64(7), object(4)
memory usage: 8.5+ KB
None


In [42]:
#Rename the column names appropriately
print(df.columns)
df = df.rename(columns={'Mat':'Match', 
                        'Inns':'Innings',
                        'Wkts':'Wickets',
                        'Ave': 'Average',
                        'Econ':'economy_rate',
                        'SR': 'Strike_Rate',
                        "4": "Four",
                        "5": "Five runs"})
display(df.head())

Index(['Player', 'Span', 'Match', 'Innings', 'Balls', 'Runs', 'Wickets',
       'Average', 'economy_rate', 'Strike_Rate', 'Four', 'Five runs',
       'Player_Name', 'Country'],
      dtype='object')


Player       Span  Match  Innings  Balls   Runs  Wickets  Average  \
0  M Muralitharan   1993-2011    350      341  18811  12326      534    23.08   
1     Wasim Akram   1984-2003    356      351  18186  11812      502    23.52   
2    Waqar Younis   1989-2003    262      258  12698   9919      416    23.84   
3      WPUJC Vaas   1994-2008    322      320  15775  11014      400    27.53   
4   Shahid Afridi   1996-2015    398      372  17670  13632      395    34.51   

   economy_rate  Strike_Rate  Four  Five runs      Player_Name        Country  
0          3.93         35.2    15         10  M Muralitharan    Asia/ICC/SL)  
1          3.89         36.2    17          6     Wasim Akram            PAK)  
2          4.68         30.5    14         13    Waqar Younis            PAK)  
3          4.18         39.4     9          4      WPUJC Vaas        Asia/SL)  
4          4.62         44.7     4          9   Shahid Afridi   Asia/ICC/PAK)

77 players played for ICC from 11 countries that is presented in this dataset.ST Jayasuriya (Asia/SL) had played for the longest period of time. BKV Prasad (INDIA),Saeed Ajmal (PAK),BAW Mendis (SL),Rashid Khan (AFG) had played for the shortest period of time.8 Australian Bowlers are present in this dataset.3 Bangladeshi players present in this dataset.RJ Hadlee (NZ)
had the lowest economy rate. MA Starc (AUS)had the lowest strike rate.Rashid Khan (AFG)had the lowest bowling average.

In [43]:
df = df.drop('Player', axis=1)
new_col_sequence = [ 'Player_Name', 'Country', 'Span', 'Match', 'Innings','Balls','Runs','Wickets', 
       'Average', 'economy_rate', 'Strike_Rate', 'Four', 'Five runs']
df = df[new_col_sequence]

display(df.head())


Player_Name        Country       Span  Match  Innings  Balls   Runs  \
0  M Muralitharan    Asia/ICC/SL)  1993-2011    350      341  18811  12326   
1     Wasim Akram            PAK)  1984-2003    356      351  18186  11812   
2    Waqar Younis            PAK)  1989-2003    262      258  12698   9919   
3      WPUJC Vaas        Asia/SL)  1994-2008    322      320  15775  11014   
4   Shahid Afridi   Asia/ICC/PAK)  1996-2015    398      372  17670  13632   

   Wickets  Average  economy_rate  Strike_Rate  Four  Five runs  
0      534    23.08          3.93         35.2    15         10  
1      502    23.52          3.89         36.2    17          6  
2      416    23.84          4.68         30.5    14         13  
3      400    27.53          4.18         39.4     9          4  
4      395    34.51          4.62         44.7     4          9